metals/src/main/scala/scala/meta/internal/parsing/FoldingRangeExtractor.scala

```diff
@@ -20,7 +20,7 @@ final class FoldingRangeExtractor(
     foldOnlyLines: Boolean,
 ) {
   private val spanThreshold = 2
-  private val distanceToEnclosingThreshold = 3
+  private val distanceToEnclosingThreshold = 1
```

Comment:
This might cause much smaller blocks to be show with folding. We should instead us the start and end of Defn, so if we hava a Block then check the parent and try to get the start of Defn and the next end marker. End markers are separate statements in the parser.